src/tcp_server.rs

```diff
@@ -1,49 +1,99 @@
+use crate::Kanata;
 use anyhow::Result;
+use net2::TcpStreamExt;
 use parking_lot::Mutex;
-use serde::Serialize;
+use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
+use std::io::Read;
 use std::net::{TcpListener, TcpStream};
+use std::str::FromStr;
 use std::sync::Arc;
+use std::time::Duration;
 
-#[derive(Debug, Serialize)]
-pub enum EventNotification {
+#[derive(Debug, Serialize, Deserialize)]
+pub enum ServerMessage {
     LayerChange { new: String },
 }
 
-impl EventNotification {
+#[derive(Debug, Serialize, Deserialize)]
+pub enum ClientMessage {
+    ChangeLayer { new: String },
```

Comment:
Added a separate ClientMessage enum here with an imperative command style for the server to handle